bugsnag-android-core/src/main/java/com/bugsnag/android/Client.java

```diff
@@ -70,11 +70,12 @@
 
     private final SessionStore sessionStore;
 
-    final SystemBroadcastReceiver systemBroadcastReceiver;
     final SessionTracker sessionTracker;
 
-    final ActivityBreadcrumbCollector activityBreadcrumbCollector;
-    final SessionLifecycleCallback sessionLifecycleCallback;
```

Comment:
Nit: these could still be declared `final` and assigned null in the constructor if the particular breadcrumb type is not enabled. This would avoid the field being reassigned internally after the `Client` has been instantiated.

Reply:
OK - I've done this. It adds quite a few else clauses, so not sure it's particularly nicer code but has the benefits of being final.